internal/kafka/internal/services/clusters.go

```diff
@@ -108,6 +110,21 @@ func (c clusterService) RegisterClusterJob(clusterRequest *api.Cluster) *apiErro
 	return nil
 }
 
+// DeregisterClusterJob deregister an existing cluster
+func (c clusterService) DeregisterClusterJob(clusterId string) *apiErrors.ServiceError {
+	cluster, err := c.FindClusterByID(clusterId)
+	if err != nil {
+		return err
+	}
+
+	updateErr := c.UpdateStatus(*cluster, api.ClusterDeprovisioning)
```

Comment:
```suggestion
	updateErr := c.UpdateStatus(api.Cluster{ClusterID: clusterID}, api.ClusterDeprovisioning)
```

related to https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1511/files#r1085014963